packages/binding-opcua/README.md

```diff
@@ -1,78 +1,167 @@
+    },
+};
+```
 
--   readproperty is set to `READ`
--   writepropery is set to `WRITE`
--   invokeaction is set to `CALL_METHOD`
+### defining a property
+
+```javascript
+const thingDescription = {
+    // ...
+    properties: {
+        temperature: {
+            description: "the temperature",
+            observable: true,
+            readOnly: true,
+            unit: "m/s",
+            type: "number",
+            forms: [
+                {
+                    href: "opc.tcp://opcuademo.sterfive.com:26543",
+                    op: ["readproperty", "observeproperty"],
+                    "opcua:nodeId": "ns=1;s=Temperature",
+                },
+            ],
+        },
+    },
+};
+```
 
-## New DataSchema Fields for the OPC UA Binding
+## Advanced
 
-OPC UA uses custom Datatypes on the wire, which requires additional translations from the JSON model used at WoT application level.
-To enable support for a proper Datatype translation, the DataSchema information has to be extended with the following terms, and the schema also be passed down to the Binding.
+The OPC-UA binding for node-wot offers additionals feature to allow you to interact with
+OPCUA Variant and DataValue in OPCUA JSON encoded form.
+For an example of use, you can dive into the unit test of the binding-opcua library.
 
-### opc:DataType
+### Exploring the unit tests
 
-Among all the possible OPC UA Datatypes, at the moment all the ones supported by [node-wot](https://github.com/node-opcua/node-opcua/blob/master/packages/node-opcua-variant/source/DataType_enum.ts) can be used.
-The binding uses a new custom ContentSerdes in order to handle the proper OPC UA Datatype and adjust the OPC UA request accordingly. Please note that this binding does not directly handle OPC UA raw bytes but instead an intermediate format to be passed to another NodeJS library ([node-opcua](https://github.com/node-opcua)) that is in charge of making the raw conversion. In this sense, the new ContentSerdes strongly depends on the Nodejs library used behind.
+A set of examples can be found in this unit test: packages\binding-opcua\test\full-opcua-thing-test.ts
 
 ## Additional tools
 
-A basic OPC UA server crawler and a basic OPC UA -> TD translator can be found [here](https://github.com/lukesmolo/wot-utils/tree/master/opcua-crawler)
+### basic OPC-UA demo server
+
+A basic demo OPC-UA server can be started using the following command.
+
+```
+thingweb.node-wot> ts-node packages/binding-opcua/test/fixture/basic-opcua-server.ts
+Server started opc.tcp://<YOURMACHINENAME>:7890
+```
+
+### awesome WoT - OPCUA tools
 
-## TODO
+the [node-wot-opcua-tools](https://github.com/node-opcua/node-wot-opcua-tools) project provides
+some useful applications built on top of thingweb.node-wob and the OPCUA binding
 
--   [ ] Subscriptions implementation (EVENTS) with Sub/Pub protocol
--   [x] TEST
--   [ ] (OPC UA Server Protocol Binding ?)
+ref: https://github.com/node-opcua/node-wot-opcua-tools
```

Comment:
Minor: I think this entire line is not needed. The link is already provided in the paragraph before that.

Reply:
OK fixed.